Objects/typeobject.c

```diff
@@ -9157,6 +9158,22 @@ releasebuffer_call_python(PyObject *self, Py_buffer *buffer)
     Py_DECREF(mv);
 }
 
+static void
+releasebuffer_call_python(PyObject *self, Py_buffer *buffer)
+{
+    // bf_releasebuffer may be called while an exception is already active.
+    // We have no way to report additional errors up the stack, because
+    // this slot returns void, so we simply stash away the active exception
+    // and restore it after the call to Python returns.
+    PyObject *type, *value, *traceback;
+    PyErr_Fetch(&type, &value, &traceback);
```

Comment:
Use newer `PyErr_GetHandledException`

Reply:
I tried that at first, but it didn't work. I think that's meant to be used only when an exception has been caught (you're in the equivalent of an `except` block), while this code is invoked when an exception was just raised.

Reply:
I meant to suggest `PyErr_GetRaisedException`, maybe I made copy paste mistake yesterday. I have pushed change to use it instead.

Reply:
Ah thanks, looks like that one is new in 3.12 and I was looking at the 3.11 docs when I wrote this. Thanks for pushing the fix!